lib/logstash/inputs/irc.rb

```diff
@@ -115,6 +115,7 @@ def run(output_queue)
         handle_response(msg, output_queue)
       rescue ThreadError
         # Empty queue
+        sleep 1;
```

Comment:
@yaauie I remember running into the same kinds of issues with Ruby's `Queue` in LS core.
This obviously fixes the issue, but is also somewhat questionable, in that we hard sleep for a second while things could be queuing up in `irc_queue` right?
What I did in core was to simply use a standard Java `java.util.concurrent.LinkedBlockingQueue` instead of the Ruby `Queue` and changed the `pop` to a `poll(timeout, timeunit)`. Couldn't we do the same thing here?